### PR TITLE
Feat tabs closable and on tab close

### PR DESCRIPTION
**Pull request by DaiQiangReal**
[中文模板 / Chinese Template](https://github.com/DouyinFE/semi-design/blob/main/.github/PULL_REQUEST_TEMPLATE.zh-CN.md)

- [ ] I have read and followed [Pull Request Guidelines](https://github.com/DouyinFE/semi-design/blob/main/CONTRIBUTING-en-US.md#pull-request-guidelines) of the contributing guide.

### What kind of change does this PR introduce? (check at least one)

 - [ ] Bugfix
 - [x] Feature
 - [ ] Code style update
 - [ ] Refactor
 - [ ] Test Case
 - [ ] TypeScript definition update
 - [ ] Document improve
 - [ ] CI/CD improve
 - [ ] Branch sync
 - [ ] Other, please describe:

### PR description

Feat tabs closable and on tab close
Fixes #

### Changelog
🇨🇳 Chinese
- 修复 ...
1.  close icon hover 样式调整为 color text-1
2. 删除了 close icon margin right 8px
3. 去除了 interface import
3. 移动 closable api 从 tab 到了 tabPlane 上
4. 新增 onTabClose Api 
5. 优化部分函数命名
---

🇺🇸 English
- Fix ...

### Checklist
- [ ] Test or no need
- [ ] Document or no need
- [ ] Changelog or no need

### CODE REVIEW

File: content/navigation/tabs/index-en-US.md [hunk 1]
```diff
@@ -535,6 +535,7 @@ tabPosition | The position of the tab, support `top` (horizontal), `left` (verti
 type | The style of the label bar, optional `line`, `card`, `button` | string | `line` |
 onChange | Callback function when switching tab pages | function(activeKey: string) | None |
 onTabClick | Click event | function(key: string, e: Event) | None |
+onTabClose | executed when tab closed by user, **>=2.0.1**  |  function(tabKey: string) | None
```
Comment (pointhalo) [inline content/navigation/tabs/index-en-US.md:538]: version 2.1.0？

File: packages/semi-ui/tabs/index.tsx [hunk 2]
```diff
@@ -106,7 +134,7 @@ class Tabs extends BaseComponent<TabsProps, TabsState> {
             setNewActiveKey: (activeKey: string): void => {
                 this.setState({ activeKey });
             },
-            setNewPanes: (panes: Array<PlainTab>): void => {
+            notifyPanesUpdate: (panes: Array<PlainTab>): void => {
                 this.setState({ panes });
```
Comment (shijiatongxue) [inline packages/semi-ui/tabs/index.tsx:138]: 为啥 notify 函数却 setState 了

File: packages/semi-ui/tabs/index.tsx [hunk 3]
```diff
@@ -192,7 +223,6 @@ class Tabs extends BaseComponent<TabsProps, TabsState> {
 
     deleteTabItem = (tabKey: string, event: MouseEvent<HTMLDivElement>) => {
         event.stopPropagation();
-        if(this.state.panes.length === 1) return
```
Comment (shijiatongxue) [inline packages/semi-ui/tabs/index.tsx:195]: 漏了分号

File: packages/semi-ui/tabs/TabBar.tsx [hunk 4]
```diff
@@ -1,14 +1,14 @@
-import React, { ReactNode, ReactElement, MouseEvent } from 'react';
+import React, { MouseEvent, ReactElement, ReactNode } from 'react';
```
Comment (shijiatongxue) [inline packages/semi-ui/tabs/TabBar.tsx:1]: 这时有啥自动化工具会自动改下位置吗？

File: packages/semi-ui/tabs/TabBar.tsx [hunk 5]
```diff
@@ -195,7 +195,7 @@ class TabBar extends React.Component<TabBarProps, TabBarState> {
     };
 
     renderOverflow = (items: [Array<OverflowItem>, Array<OverflowItem>]): Array<ReactNode> => items.map((item, ind) => {
-        const icon = ind === 0 ? <IconChevronLeft /> : <IconChevronRight />;
+        const icon = ind === 0 ? <IconChevronLeft/> : <IconChevronRight/>;
```
Comment (shijiatongxue) [inline packages/semi-ui/tabs/TabBar.tsx:198]: 按理说有个空格是符合规范的？

File: content/navigation/tabs/index.md [hunk 6]
```diff
@@ -518,22 +518,33 @@ import React from 'react';
 import { Tabs, TabPane } from '@douyinfe/semi-ui';
 
 class App extends React.Component {
```
Comment (shijiatongxue) [inline content/navigation/tabs/index.md:520]: 建议下次我们的新 demo 都用 function 写法 o(╥﹏╥)o

File: packages/semi-ui/tabs/index.tsx (hunk 2 above)
Comment (shijiatongxue) [reply packages/semi-ui/tabs/index.tsx:138]: 这个函数有用到吗？看起来没有地方调用

**Pull request merged**